### PR TITLE
Changed the validator to return false for a null email input #4108

### PR DESCRIPTION
**Pull request by rbhatta99**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #4108  email validator not working on sign up

## Pull Request Checklist

- [x] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/4.6.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: doc/sphinx-guides/source/_static/installation/files/root/big-data-support/checksumValidationSuccess.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "status": "validation passed",
-  "uploadFolder": "DNXV2H",
-  "totalSize": 1234567890
+  "uploadFolder": "OS7O8Y",
+  "totalSize": 72
```
Comment (ferrys) [inline doc/sphinx-guides/source/_static/installation/files/root/big-data-support/checksumValidationSuccess.json]: What is this change needed for?

Comment (rbhatta99) [reply doc/sphinx-guides/source/_static/installation/files/root/big-data-support/checksumValidationSuccess.json]: oh, this is rsync related stuff. the file there is used for the mock dcm setup. And I had to change the dataset identifier to point to the one I'm currently using. This can be ignored. I added it by mistake!

Comment (djbrooke) [reply doc/sphinx-guides/source/_static/installation/files/root/big-data-support/checksumValidationSuccess.json]: @rbhatta99 @ferrys OK if we move to QA or does this need to be removed?

Comment (rbhatta99) [reply doc/sphinx-guides/source/_static/installation/files/root/big-data-support/checksumValidationSuccess.json]: This will not matter for QA at all, as this is just for DCM purposes. But, don't out this into QA just yet. I got some feedback from @sekmiller and I'm trying something out.

**Comment by matthew-a-dunlap** [comment]
Discussed with @sekmiller (after his discussions with @rbhatta99 ) and the logic to only return null if email is required makes sense. In the default installation this doesn't matter but this is more futureproof/amenable to customization.

**Comment by matthew-a-dunlap** [comment]
Discussed with @sekmiller (after his discussions with @rbhatta99 ) and the logic to only return null if email is required makes sense. In the default installation this doesn't matter but this is more futureproof/amenable to customization.
